pkg/filter/network/grpc/factory.go

```diff
@@ -220,14 +221,14 @@ type Handler struct {
 }
 
 // New a grpc server with address. Same address returns same server, which can be start only once.
-func (s *Handler) New(addr string, conf json.RawMessage, options ...grpc.ServerOption) (*registerServerWrapper, error) {
```

Comment:
It seems a break change, how about providing another method likes `NewUnix` (just an example name)?

Reply:
I checked the code of Mosn. The changes of these two functions will not cause compatibility problems. Adding new functions will lead to code duplication and redundancy. I personally think that it may be more reasonable to add a parameter if there is no compatibility problem. 😊

Reply:
But there has an special using case: users could use mosn as package, like layotto, without mosnd.

As for code duplication and redundancy, we could use a single method for them all, the exposed methods are just adaptor for your functions

Reply:
> I checked the code of Mosn. The changes of these two functions will not cause compatibility problems. Adding new functions will lead to code duplication and redundancy. I personally think that it may be more reasonable to add a parameter if there is no compatibility problem. 😊

Add an unexport function that have network.
 New() wraps tcp, NewUnix wraps unix.
No code duplication

Reply:
Oooops，there is some lint failes

```
  string `unix` has 3 occurrences, make it a constant (goconst)
  
  Raw Output:
  filter/network/grpc/factory.go:88:16: string `unix` has 3 occurrences, make it a constant (goconst)
  	if network == "unix" {
  	              ^
  Error: [golangci] reported by reviewdog 🐶
  variable 'network' is only used in the if-statement (filter/network/grpc/factory.go:88:2); consider using short syntax (ifshort)
  
  Raw Output:
  filter/network/grpc/factory.go:76:2: variable 'network' is only used in the if-statement (filter/network/grpc/factory.go:88:2); consider using short syntax (ifshort)
  	network := cfg.Network
  	^
```

Unix and tcp could define a new type and mark them as const enums

Reply:
fixed